src/pages/console/formbuilder/lifecycle.mdx

```diff
@@ -0,0 +1,165 @@
+export const meta = {
+  title: "Manage form lifecycle",
+  description: ""
+}
+
+Hook into the form's lifecycle events to customize user input before submission, run validations, handle errors, or self-manage user input events. 
+
+![Lifecycle diagram of the Amplify Forms](/images/console/formbuilder/lifecycle-diagram.png)
+
+|No.|Event|Description|Use cases|
+|-|-|-|-|
+|1|Form is mounted|The inputs are either empty or pre-populated based on a default value provided by you.|If your user clicks on the **Clear** or **Reset** button, they'll be brought back to this state.|
+|2|[`onChange`](#get-form-data-as-your-user-inputs-data---onchange)|Event when form data is changed by the user.|Use this to get the form data after every user input.
+|3|[`onValidate`](#extend-validation-rules-in-code---onvalidate)| Event hook for custom validations. This event triggers after `onChange`.| Use this to extend validation rules via code. `onValidate` also supports asynchronous validation rules, which enable you to validate the form input against external APIs.|
+|4|[`onSubmit`](#handle-form-data-submissions---onsubmit)| Event when your user clicks the **Submit** button.|  - If your form is **not connected to a data model**, use set this event handler to retrieve the form data <br /> - If your form is **connected to a data model**, use this to customize the provided form data before they are saved to the cloud. |
+|5|[`onSuccess`](#handle-form-data-successfully-saving-to-the-cloud---onsuccess)| Event when saving form data to the cloud succeeds.| Use this to dismiss the form or reroute your user after a successful form submission. Only use this if your form is connected to a data model.|
+|6|[`onError`](#handle-form-submission-errors---onerror)| Event when saving form data to the cloud fails.| Use this to log the error and investigate further if your validation rules need to be enhanced to catch input formatting issues. Only use this if your form is connected to a data model.|
+|7|[`onCancel`](#handle-user-clicking-on-undefined-action-button---oncancel)| Event when your user clicks on the **Cancel** button.| Use this to dismiss the form without saving the form data.|
+
+## Get form data as your user inputs data - onChange
+
+In some cases, you want to get the form data in real-time as the user is filling the form. The `onChange` event provides you the form data in the `fields` parameter. We recommend you to use `onChange` if you opted to [hide all form action buttons](/console/formbuilder/call-to-action/#toggle-visibility-for-submit-cancel-clear-and-reset-buttons) (Submit, Cancel, Clear, Reset).
+
+**Note:** Usually, you only need to set the `onSubmit` to get the form data when the user clicks on **Submit**.
```

Comment:
I don't understand this section. Why are we discussing `onSubmit` in the section for `onChange`?